tests/test/projection.rs

```diff
@@ -457,6 +457,48 @@ fn normalize_gat2() {
     }
 }
 
+#[test]
+fn normalize_gat_const() {
+    test! {
+        program {
+            trait StreamingIterator<T> { type Item<const N>; }
+            struct Span<const N, T> { }
+            struct StreamIterMut<T> { }
+            impl<T> StreamingIterator<T> for StreamIterMut<T> {
+                type Item<const N> = Span<N, T>;
+            }
+        }
+
+        goal {
+            forall<const N, T> {
+                exists<U> {
+                    Normalize(<StreamIterMut<T> as StreamingIterator<T>>::Item<N> -> U)
+                }
+            }
+        } yields {
+            "Unique; substitution [?0 := Span<!1_0, !1_1>], lifetime constraints []"
```

Comment:
I guess we don't have any sigil for "constant placeholders"? Seems consistent, I suppose